lzy/allocator/src/main/java/ai/lzy/allocator/alloc/UnmountDynamicDiskAction.java

```diff
@@ -94,78 +104,72 @@ private StepResult prepareActiveMounts() {
         return StepResult.CONTINUE;
     }
 
-    private StepResult recreateMountPod() {
-        if (vm == null) {
+    private StepResult getNextMountPodId() {
+        if (vmIsDeleting()) {
             return StepResult.ALREADY_DONE;
         }
-
-        if (updatedMountPod != null) {
+        if (nextId != null) {
             return StepResult.ALREADY_DONE;
         }
-
-        var mountPodName = vm.instanceProperties().mountPodName();
-        if (mountPodName == null) {
+        if (!dynamicMount.mounted()) {
+            return StepResult.ALREADY_DONE;
+        }
+        if (activeMounts == null || activeMounts.isEmpty()) {
             return StepResult.ALREADY_DONE;
         }
 
-        var dynamicMounts = activeMounts.stream()
-            .filter(x -> !x.id().equals(dynamicMount.id()))
-            .toList();
-        log().info("{} Recreating mount pod {} with {} mounts", logPrefix(), mountPodName, dynamicMounts.size());
         try {
-            updatedMountPod = allocationContext.mountHolderManager().recreateWith(vm.spec(),
-                ClusterPod.of(dynamicMount.clusterId(), mountPodName), dynamicMounts);
-        } catch (KubernetesClientException e) {
-            log().error("{} Failed to detach volume {} from pod {}", logPrefix(), dynamicMount.mountName(),
-                mountPodName, e);
-            if (KuberUtils.isNotRetryable(e)) {
-                return StepResult.CONTINUE;
-            }
-            return StepResult.RESTART;
+            nextId = withRetries(log(), () -> allocationContext.vmDao().getNextMountPodId(vm.vmId(), null));
+        } catch (Exception e) {
+            log().error("{} Couldn't get next mount id for vm", logPrefix(), e);
+            return StepResult.CONTINUE;
```

Comment:
why `CONTINUE`?
I'd expect either `ALREADY_DONE` or `RESTART`

Reply:
I assumed that if we changed the state of our operation (table `vm`), then we should return `CONTINUE` so that we can check whether our operation is still valid (or already canceled). If in the current step we don't modify our state (table `vm`), then it is also not necessary to modify the `operation` table.

But this can be considired as optimization of unnecessary writes to the `operation` table.